Update documentation + add convenience APIs (#242)
segmentio/parquet-go: README.md

go get github.com/segmentio/parquet-go
```

> Go 1.17 is required to use the package, but some APIs and optimizations are only

Reviewer: ```suggestion
Go 1.17 is the lowest supported version. Some APIs and optimizations are only
```

Reviewer: Previously you might think you _had_ to use 1.17 - saying it's "required" doesn't make sense in context if you are using 1.18

Author: I can rephrase 👍

the schemas, there are no type conversions performed, nor ways to rename
columns, etc... More advanced conversion rules may be added in the future.

### Sorting Row Groups: [parquet.GenericBuffer[T]](https://pkg.go.dev/github.com/segmentio/parquet-go#Buffer)

The `parquet.GenericWriter[T]` type is optimized for minimal memory usage,

> keeping the order rows unchanged and flushing pages as soon as they are filled.

Reviewer: ```suggestion
keeping the order of rows unchanged and flushing pages as soon as they are filled.
```

Parquet supports expressing columns by which rows are sorted through the
declaration of _sorting columns_ on row groups. Sorting row groups requires
buffering all rows before ordering and writing them to a parquet file.

To help with those use cases, the `segmentio/parquet-go` package exposes the
`parquet.GenericBuffer[T]` type which acts as a buffer of rows and implements
`sort.Interface` to allow applications to sort rows prior to writing them
to a file.

The columns that rows are ordered by are configured when creating
`parquet.GenericBuffer[T]` instances using the `parquet.SortingColumns` function
to construct row group options configuring the buffer. The type of parquet
columns defines how values are compared, see [Parquet Logical Types](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md)
for details.

When written to a file, the buffer is materialized into a single row group with
the declared sorting columns. After being written, buffers can be reused by
calling their `Reset` method.

The following example shows how to use a `parquet.GenericBuffer[T]` to order rows
written to a parquet file:

```go

> buffer := parquet.NewGenericBuffer[RowType](

Reviewer: Is the user supposed to define RowType? 

If so I'd maybe define it as part of the example - it's better to be explicit than implicit for docs, and to show the relationship between the objects.

Author: Yes, `RowType` is whatever type the application wants to use since it's a generic.

Which do you think would be better:

- define `RowType` once in the first example
- define it every time (maybe fills the example with a lot of redundant and not very relevant info)
- use a more generic name like `[T]` of `[...]` to specify that this part of the code is not relevant to the example

of Go values:

```go

> func writeColumns(buffer *parquet.GenericBuffer[RowType], firstNames []string) error {

Reviewer: Same comment here re: RowType
